core/src/main/scala/org/broadinstitute/dsde/rawls/util/WithSupport.scala

```diff
@@ -41,16 +42,28 @@ trait MethodWiths {
     }
   }
 
-  def withMethodInputs[T](methodConfig: MethodConfiguration, userInfo: UserInfo)(op: (String, Seq[MethodInput]) => ReadWriteAction[T])(implicit executionContext: ExecutionContext): ReadWriteAction[T] = {
+  def withMethodInputs[T](methodConfig: MethodConfiguration, userInfo: UserInfo)(op: (String, Seq[MethodInput], Seq[MethodInput]) => ReadWriteAction[T])(implicit executionContext: ExecutionContext): ReadWriteAction[T] = {
     // TODO add Method to model instead of exposing AgoraEntity?
     val methodRepoMethod = methodConfig.methodRepoMethod
     withMethod(methodRepoMethod.methodNamespace, methodRepoMethod.methodName, methodRepoMethod.methodVersion, userInfo) { method =>
-      withWdl(method) { wdl => MethodConfigResolver.gatherInputs(methodConfig,wdl) match {
-        case Failure(exception) => DBIO.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.BadRequest, exception)))
-        case Success(methodInputs) => op(wdl,methodInputs)
-      }}
+      withWdl(method) { wdl =>
+        MethodConfigResolver.gatherInputs(methodConfig, wdl) match {
+          case Failure(exception) => DBIO.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.BadRequest, exception)))
+          case Success(methodInputs) =>
+            //Remove inputs that are both empty and optional
+            val (emptyOptionalInputs, inputsToProcess) = methodInputs.partition(input => input.workflowInput.optional && input.expression.isEmpty)
```

Comment:
This partition was in `withSubmissionParameters()` but I felt it made more sense here